src/librustdoc/html/layout.rs

```diff
@@ -158,7 +158,9 @@ pub fn render<T: Print, S: Print>(
         keywords = page.keywords,
         favicon = if layout.favicon.is_empty() {
             format!(
-                r#"<link rel="shortcut icon" href="{static_root_path}favicon{suffix}.ico">"#,
+                r##"<link rel="alternate icon" type="image/png" href="{static_root_path}favicon-16x16{suffix}.png">
+<link rel="alternate icon" type="image/png" href="{static_root_path}favicon-32x32{suffix}.png">
+<link rel="icon" type="image/svg+xml" href="{static_root_path}favicon{suffix}.svg">"##,
```

Comment:
Shouldn't the order be the "icon" first and the "alternate" then?

Reply:
Hmm, I'm not sure? It was in this order in the original PR, and looking at [this page](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link) it would appear that `alternate` comes first:

```
<link href="default.css" rel="stylesheet" title="Default Style">
<link href="fancy.css" rel="alternate stylesheet" title="Fancy">
<link href="basic.css" rel="alternate stylesheet" title="Basic">
```

But I'm not sure if there's a spec somewhere for it or if it even matters.

Reply:
From the code you showed, non-alternate comes first. ;)

Reply:
Oh sorry, my bad! I thought you were referring to the ordering of the entries in the `rel` string :joy:

I'll move the `icon` only link to the top later today 👍

Reply:
Thanks! ;)

Reply:
Sorry, I got busy yesterday. Fixed :)